src/raft/uv.h

```diff
@@ -47,6 +47,7 @@ typedef unsigned long long uvCounter;
 /* Information persisted in a single metadata file. */
 struct uvMetadata
 {
+	uint64_t format_version;
```

Comment:
What about naming the enum and using the type here instead of the generic uint64_t?

Reply:
Mainly avoiding casts. The value has to be represented as `uint64_t` at the lowest level since that's what the encoding/decoding functions work with, and life is simpler if we keep it the same type everywhere. I also have a slight aversion to named enum types because I think using a normal integer type reminds you to check for out-of-range values. (On the other hand, I think cc has some special diagnostics that only kick in if you `switch` on a value of enum type.)

Reply:
I am not sure about this. I see your point about having to remember to check that the int is on the range of the enum though. Couldn't we make it so untrusted sources return an integer but when we define functions / structs that expect a valid value we use the enum? That way, when passing the enum you get reminded to check for bounds before casting and we also get the benefits of having more type information.